### PR TITLE
Added some Morse Code code

### PR DESCRIPTION
**Pull request by jskrist**
This PR includes [a tree implementation of Morse Code](https://en.wikipedia.org/wiki/Morse_code#Alternative_display_of_common_characters_in_International_Morse_code) and a new function to convert a text message into an equivalent morse code representation.

Added new tests for this functionality and passes all existing tests as well.

### CODE REVIEW

**Comment by taldcroft** [comment]
Interesting... is this your entry for a code obfuscation contest?  :smiley:

File: play/jamesk.py [hunk 1]
```diff
@@ -30,3 +30,90 @@ def knockknock():
...
+                # pop off the next node in the queue to process
+                this_node = node_queue.pop(0)
+                # add the current character to the left of the current node
+                this_node.left = Node(char)
+
+
+    def find_char(self, char, this_node=None, morse_str = ""):
+
+        # "*" is used in place of invalid characters, so we cannot match it correctly
+        if(char == "*"):
```
Comment (taldcroft) [inline play/jamesk.py:70]: No extra parentheses. Should be:
```
if char == "*":
```

Comment (taldcroft) [reply play/jamesk.py:70]: Try running:
```
flake8 . --count --exclude=docs --ignore=E402,W503,F541 --max-line-length=100 --show-source --statistics
```
See https://occweb.cfa.harvard.edu/twiki/bin/view/Aspect/SkaPython#Check_and_maintain_PEP8_and_flak

Comment (taldcroft) [reply play/jamesk.py:70]: Or better yet set up VS code to enable Python linting with flake8. Then all those problems show up immediately.
image

**Comment by jskrist** [comment]
This made me smile 😄 
Nope, I just thought it would be fun to implement a simple tree and depth-first recursive search algorithm in python 🙃

File: play/jamesk.py (hunk 1 above)
Comment (jskrist) [reply play/jamesk.py:70]: Thanks Tom.  I installed the flake8 linting into VS Code and made some updates.  Interestingly, the parenthesis for the if statements were not flagged by either VS Code or running the flake8 command directly, so I left them as-is for now.

Comment (taldcroft) [reply play/jamesk.py:70]: Interesting that this is not actually in PEP8. But I've never seen "production" Python code that looks like that. See e.g. the standard library code for `pathlib` https://github.com/python/cpython/blob/master/Lib/pathlib.py. People do sometimes use parens in a longer if statement to improve clarity.

**Review by taldcroft — APPROVED** [review]
Looks good!

**Pull request merged**